### PR TITLE
Remove SHOPT_COSHELL

### LINKED ISSUE #619: Remove the coshell facility
We're not actually building the `coshell` command; i.e., the coshell service daemon. We're building just the coshell library and the bits integrated into `ksh`. The facility provided by the coshell facility isn't really central to the korn shell. And most companies (and individuals) already have similar solutions. It also seems to be tightly integrated with ksh rather than being shell agnostic. Lastly, and most importantly, it doesn't look like coshell is secure. Note that its *PROMO.mm* file mentions `rsh` as an alternative and does not mention `ssh`. This code dates back to an era when security concerns were very different than today. I can't imagine any company with even a minimally competent security team using this. I'd like to drop the coshell code.

### PR DESCRIPTION
**Pull request by krader1961**
This removes libcoshell and the ksh code that depends on it. Since we
already removed the `coshell` server there is no need for ksh to be able
to talk to it.

Resolves #619

### CODE REVIEW

File: src/cmd/ksh93/meson.build [hunk 1]
```diff
@@ -26,7 +25,6 @@ shared_c_args += [
     '-D_BLD_shell',
     '-DERROR_CONTEXT_T=Error_context_t',
     '-DKSHELL',
-    '-DSHOPT_COSHELL',
```
Comment (siteshwar) [inline src/cmd/ksh93/meson.build]: This option and related code was enabled in the last stable release. I suspect there are going to be side effects for removing this code.

Comment (krader1961) [reply src/cmd/ksh93/meson.build]: Obviously there shouldn't be any side-effects of removing, or simply disabling, this feature if is truly is optional as implied by the build time `SHOPT_COSHELL` symbol. Other than making it impossible to talk to a coshell server. And honestly a) I would be surprised if this feature still works in light of all the other changes, and b) I would be even more surprised if there is anyone using this feature.

I'm willing to change this to just build without the feature enabled by default plus the fix to `job_walk()` I mentioned in my previous comment and a deprecation warning in *CHANGELOG.md*. That way if a distro like Red Hat believes there are users of this facility they can still enable it.

Comment (siteshwar) [reply src/cmd/ksh93/meson.build]: I would prefer to just disable this feature instead of removing associated code.

**Comment by krader1961** [comment]
Closing in favor of the more conservative #649.

**Pull request closed without merging**